### PR TITLE
Add entry for libp2p-core vulnerability

### PR DESCRIPTION
**Pull request by thomaseizinger**
I am a maintainer of this crate and we have already published a fix for it. Opening this advisory so folks get notified of the vulnerability and can upgrade quickly.

### CODE REVIEW

**Comment by Shnatsel** [comment]
Do I understand correctly that the issue first appeared in `0.30.0`? There should be an `unaffected` field to indicate this, then (the data from function definitions currently isn't automatically propagated).

File: crates/libp2p-core/RUSTSEC-0000-0000.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+id = "RUSTSEC-0000-0000"
+package = "libp2p-core"
+date = "2022-02-07"
+categories = ["crypto-failure"]
+
+[affected]
+functions = { "libp2p_core::PeerRecord::from_signed_envelope" = [">= 0.30.0"] }
+
+[versions]
+patched = [">= 0.31.1"]
```
Comment (Shnatsel) [inline crates/libp2p-core/RUSTSEC-0000-0000.md:12]:
```suggestion
unaffected = ["< 0.30.0-rc.1"]
patched = [">= 0.31.1"]
```

File: crates/libp2p-core/RUSTSEC-0000-0000.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
+```toml
+[advisory]
+id = "RUSTSEC-0000-0000"
+package = "libp2p-core"
+date = "2022-02-07"
+categories = ["crypto-failure"]
+
+[affected]
+functions = { "libp2p_core::PeerRecord::from_signed_envelope" = [">= 0.30.0"] }
```
Comment (Shnatsel) [inline crates/libp2p-core/RUSTSEC-0000-0000.md:9]:
```suggestion
functions = { "libp2p_core::PeerRecord::from_signed_envelope" = [">= 0.30.0-rc.1"] }
```

**Review by Shnatsel — COMMENTED** [review]
I assume the release candidate was also affected?

**Comment by thomaseizinger** [comment]
> I assume the release candidate was also affected?

Ah yes. I forgot that we had a RC.

> Do I understand correctly that the issue first appeared in `0.30.0`?

Yes, that is when the entire feature got first added: https://github.com/libp2p/rust-libp2p/blob/master/core/CHANGELOG.md#0300-2021-11-01

**Comment by thomaseizinger** [comment]
Thanks @Shnatsel ! Do you want me to squash those commits or was applying the suggestion fine?

**Comment by Shnatsel** [comment]
You don't have to do that manually -  we've set up Github to squash before merging automatically.

**Comment by Shnatsel** [comment]
Looks good. Merging, thanks!

Also, I see that there is no semver-compatible fix for users of 0.30.x; is it possible to backport the fix to 0.30.x so that people would just get the fix with a `cargo update`, without any code changes?

**Comment by thomaseizinger** [comment]
> Looks good. Merging, thanks!
> 
> Also, I see that there is no semver-compatible fix for users of 0.30.x; is it possible to backport the fix to 0.30.x so that people would just get the fix with a `cargo update`, without any code changes?

I will look into that, good idea!

**Pull request merged**